srml/contract/src/lib.rs

```diff
@@ -14,41 +14,63 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate. If not, see <http://www.gnu.org/licenses/>.
 
-//! Smart-contract module for runtime; Allows deployment and execution of smart-contracts
-//! expressed in WebAssembly.
+//! # Contract Module
+//! 
+//! The contract module provides functionality to deploy and execute smart-contracts expressed in WebAssembly.
+//! To use it in your module, you need to implement the contract [`Trait`].
+//! The supported dispatchable functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Overview
+//! 
+//! This module provides the ability to create smart-contract accounts and send them messages.
+//! A smart-contract is represented by an account with associated code and storage. 
+//! When an account receives a message, the associated code gets executed.
+//! This code can alter the storage entries of the associated account, create new smart-contracts, or send messages to existing smart-contracts.
 //!
-//! This module provides an ability to create smart-contract accounts and send them messages.
-//! A smart-contract is an account with associated code and storage. When such an account receives a message,
-//! the code associated with that account gets executed.
+//! Senders must specify a gas limit with every transaction, as all actions invoked by the smart-contract require gas spend.
+//! Unused gas is refunded after the transaction, regardless of the execution outcome. 
+//! If the gas limit is reached, then all changes made at the specific transaction level are reverted (including balance transfers).
+//! 
+//! **NOTE:** Transaction failures are typically not cascading. For example, if contract A calls B and B errors
+//! somehow, A can still decide if it should proceed or error.
+//! 
+//! Finally, when the `staking` module determines an account is dead (i.e. account balance fell below the
+//! existential deposit), it reaps the account. This will delete the associated code and storage of the account.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable functions ([`Call`])
+//!  
+//! * `put_code` - Stores code in the contract account storage.
```

Comment:
```suggestion
//! * `PUT_CODE` - Stores the given binary Wasm code into the chains storage and returns its `codehash`.
```

Reply:
this is definitely better. will be more through next time!
will also change this inline for the fn itself

Reply:
btw @Robbepop are we capitalizing all function names? or just using this to distinguish dispatchable functions (vs public functions which remain lower cased)?

Reply:
I wonder if you can actually say that it returns anything, since dispatchable calls are always `Result<(), &'static str>` and can't return anything. However, they indeed signal the value of code hash by depositing an event.

Reply:
I'm going to go ahead an unresolve this conversation to add some visibility to this.